elastic/kibana
# Delete connector usage reporting task in on-prem and cloud

Towards: https://github.com/elastic/response-ops-team/issues/209

`connector_usage_reporting` task is supposed to work only in serverless.
As there is no check for that, even though it reports nothing, It was working for on-prem and cloud as well.
This PR deletes the `connector_usage_reporting` task after the first run by checking if `plugins.cloud.serverless.projectId` is missing.

### To verify:
Run Kibana on this branch, then check the reporting task in the console by running the below query.
There shouldn't be any task.

```
GET .kibana_task_manager_*/_search
{
  "query": {
    "prefix": {
      "task.taskType": {
        "value": "actions:"
      }
    }
  }
}
```

Then remove or comment out the line (`shouldDeleteTask: true`) added by this PR and save the file, Kibana will restart and register the task again. You can use the above query again to see that the task is still registered.
